src/libraries/System.Threading.Tasks.Dataflow/tests/Dataflow/TransformManyBlockTests.IAsyncEnumerable.cs

```diff
@@ -0,0 +1,663 @@
+                tb2.Post(i);
+            }
+
+            await Assert.ThrowsAsync<InvalidCastException>(() => tb1.Completion);
+            await Assert.ThrowsAsync<FormatException>(() => tb2.Completion);
+
+            Assert.True(tb1.InputCount == 0 && tb1.OutputCount == 0);
+        }
+
+        private async IAsyncEnumerable<int> ExceptionAfterAsync(int iterations)
+        {
+            for (int i = 0; i < iterations; i++)
+            {
+                await Task.Yield();
+                yield return i;
+            }
+            throw new FormatException();
+        }
+
+        [Fact]
+        public async Task TestFaultingAndCancellationAsyncEnumerable()
+        {
+            foreach (bool fault in DataflowTestHelpers.BooleanValues)
+            {
+                var cts = new CancellationTokenSource();
+                var tb = new TransformManyBlock<int, int>(DataflowTestHelpers.ToAsyncEnumerable, new ExecutionDataflowBlockOptions { CancellationToken = cts.Token });
+                tb.PostRange(0, 4);
+                Assert.Equal(expected: 0, actual: await tb.ReceiveAsync());
+                Assert.Equal(expected: 1, actual: await tb.ReceiveAsync());
+
+                if (fault)
+                {
+                    Assert.Throws<ArgumentNullException>(() => ((IDataflowBlock)tb).Fault(null));
+                    ((IDataflowBlock)tb).Fault(new InvalidCastException());
+                    await Assert.ThrowsAsync<InvalidCastException>(() => tb.Completion);
+                }
+                else
+                {
+                    cts.Cancel();
+                    await Assert.ThrowsAnyAsync<OperationCanceledException>(() => tb.Completion);
+                }
+
+                Assert.Equal(expected: 0, actual: tb.InputCount);
+                Assert.Equal(expected: 0, actual: tb.OutputCount);
+            }
+        }
+
+        [Fact]
+        public async Task TestCancellationExceptionsIgnoredAsyncEnumerable()
+        {
+            static IAsyncEnumerable<int> body(int i)
+            {
+                if ((i % 2) == 0) throw new OperationCanceledException();
+                return DataflowTestHelpers.ToAsyncEnumerable(i);
+            }
+
+            TransformManyBlock<int, int> t = new TransformManyBlock<int, int>(body);
+
+            t.PostRange(0, 2);
+            t.Complete();
+            for (int i = 0; i < 2; i++)
+            {
+                if ((i % 2) != 0)
+                {
+                    Assert.Equal(expected: i, actual: await t.ReceiveAsync());
+                }
+            }
+
+            await t.Completion;
+        }
+
+        [Fact]
+        public async Task TestYieldingNoResultsAsyncEnumerable()
+        {
+            foreach (int dop in new[] { 1, Environment.ProcessorCount })
+                foreach (int boundedCapacity in new[] { DataflowBlockOptions.Unbounded, 1, 2 })
+                {
+                    const int Modes = 3, Iters = 100;
```

Comment:
```suggestion
                    const int modes = 3, iters = 100;
```
private constants should be written in lower case

Reply:
No. See Point 12 what you have linked above --> PascalCase for `const` (all).